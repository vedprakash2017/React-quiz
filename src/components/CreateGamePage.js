import React from 'react';
import { connect } from 'react-redux';
import { socket } from '../app';
import { setRoom } from '../actions/game';
import { Redirect } from 'react-router-dom';
import Fade from 'react-reveal/Fade';

export class CreateGamePage extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            room: "",
            category: "0",
            difficulty: "any",
            questionCount: "5",
            error: "",
            background: ""
        }
    }

    onRoomChange = (e) => {
        const room = e.target.value;
        this.setState({ room });
    };

    onCategoryChange = (e) => {
        const category = e.target.value;
        this.setState({ category })
    }

    onDifficultyChange = (e) => {
        const difficulty = e.target.value;
        this.setState({ difficulty });
    }

    onCountChange = (e) => {
        const questionCount = e.target.value;
        this.setState({ questionCount });
    }
    submitForm = (e) => {
        e.preventDefault();
        const config = {
            room: this.state.room,
            category: this.state.category,
            difficulty: this.state.difficulty,
            questionCount: this.state.questionCount
        };
        //console.log("submitting")
        socket.emit("createRoom", config, (res) => {
            //console.log("res!", res);
            if (res.code === "success") {
                this.setState({ error: "" })
                this.props.setRoom(this.state.room);
                this.props.history.push("/lobby");
            } else {
                this.setState({ error: res.msg })
            }
        });

    };

    render() {
        return (
            <div className="content-container" style={{height:"90vh"}}>
                {
                    this.props.type === "" && <Redirect to="/" />
                }
                <div className="box-layout__box" style={{ background: "rgba(0, 0, 0, 0.1)"}}> 
                    <Fade>
                        <form className="form" onSubmit={this.submitForm}>
                            <h1 className={"box-layout__title"}>Create New Game</h1>
                            {this.state.error && <p className="form__error">{this.state.error}</p>}
                            <input
                                type="text"
                                placeholder="Room Name"
                                autoFocus
                                value={this.state.room}
                                onChange={this.onRoomChange}
                                className="text-input"
                            />
                            <select className="select" value={this.state.category} onChange={this.onCategoryChange}>
                                <option key={"0"} value={"0"}>Any Categories</option>
                                {
                                    this.props.categories.map((category) => {
                                        return <option key={category.id} value={category.id}>{category.name}</option>
                                    })
                                }
                            </select>
                            <select className="select" value={this.state.difficulty} onChange={this.onDifficultyChange}>
                                <option key={"any"} value={"any"}>Any Difficulty</option>
                                <option key="easy" value="easy">Easy</option>
                                <option key="medium" value="medium">Medium</option>
                                <option key="hard" value="hard">Hard</option>
                            </select>
                            <select className="select" value={this.state.questionCount} onChange={this.onCountChange}>
                                <option key="5" value="5">5 Questions</option>
                                <option key="10" value="10">10 Questions</option>
                                <option key="15" value="15">15 Questions</option>
                            </select>

                            
                            <div class="btn">
                                            <button class="btn-hover color-9" onClick={this.handleReset}>Create</button>
                            </div>
                            {/* <button className="button">Create</button> */}

                        </form>
                    </Fade>

                </div>
            </div>
        )
    }
}


const mapStateToProps = (state) => ({
    categories: state.game.categories,
    type: state.type
});

const mapDispatchToProps = (dispatch) => ({
    setRoom: (room) => dispatch(setRoom(room))
});

export default connect(mapStateToProps, mapDispatchToProps)(CreateGamePage);