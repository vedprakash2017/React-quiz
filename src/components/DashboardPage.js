import React ,{ useState, useEffect} from 'react';
import { connect } from 'react-redux';
import { setHost, setPlayer } from '../actions/clientType';
import Fade from 'react-reveal/Fade';
export class DashboardPage extends React.Component {
        state = {
          left: 0,
          top: 0,
          play: false,
          audio : new Audio("https://edolve.s3.ap-south-1.amazonaws.com/1620296923Adventure Trailer.mp3")
      
        }
      
        componentDidMount() {
          // When the component is mounted, add your DOM listener.
        document.addEventListener("mousemove", this.handleCursor);
        this.state.audio.addEventListener('ended', () => this.setState({ play: true }));
        if(this.state.play == true)
        {
        this.state.audio.play();
        // this.state.play = false;
        this.state.audio.loop = true;  
        this.state.audio.volume = 0.3;
        // this.togglePlay();
        }
    }
        componentDidUpdate(){
            if(this.state.play == true)
        {this.state.audio.play();
        this.state.audio.loop = true;  
        this.state.audio.volume = 0.3;
        }
    }
        componentWillUnmount() {
          // Make sure to remove the DOM listener when the component is unmounted.
          document.removeEventListener("mousemove", this.handleCursor);

        this.state.audio.removeEventListener('ended', () => this.setState({ play: false }));
        this.state.audio.pause();
        this.state.play = false;

        }
            togglePlay = () => {
        this.setState({ play: !this.state.play }, () => {
          this.state.play ? this.state.audio.play() : this.state.audio.pause();
          this.state.audio.loop = true;
        this.state.audio.volume = 0.3;
        });
      }

        handleCursor = (e) => {
          const card = document.getElementsByClassName('card')[0];     
        //   cursor.setAttribute('style','top:'+e.pageY+'px; left:'+ e.pageX+'px;');
            if(this.state.play == false) { this.setState({ play: true });}

          var ax = -($(window).innerWidth()/2- e.pageX)/20;
          var ay = ($(window).innerHeight()/2- e.pageY)/10;
          card.setAttribute("style", "height:500px;width:475px;transform: rotateY("+ax+"deg) rotateX("+ay+"deg);-webkit-transform: rotateY("+ax+"deg) rotateX("+ay+"deg);-moz-transform: rotateY("+ax+"deg) rotateX("+ay+"deg)");

        }

    startAsHost = () => {
        this.props.setHost();
        // this.props.history.push("/create_quiz");
        this.props.history.push("/create");
    };

    startAsPlayer = () => {
        this.props.setPlayer();
        this.props.history.push("/join");
    };


    render() {
        return (


<div style={{background:"white"}}>
    {/* {  this.state.play ? <i class="fa fa-volume-up fa-3x" onClick={this.togglePlay} aria-hidden="true"></i> : <i onClick={this.togglePlay} class="fa fa-volume-mute fa-3x" aria-hidden="true"></i>} */}
    {/* <button onClick={this.togglePlay} > ON </button> */}
    <div class="main" >
        <div class="imgi"> </div> 
        <div class="card" style={{height:"500px",width:"475px"}}> 
            <div class="card-content">
            <h1>OpenTrivia</h1>
            <hr></hr>
            <p>
            A party trivia game in your web browser.
            No downloading apps just create a game on the big screen and join up with your mobile device to play
            </p>
            </div>
        </div>

        <div class="m">
            <div class="container">
                <div class="chevron"></div>
                <div class="chevron"></div>
                <div class="chevron"></div>
            </div>
        </div>
    </div>
    <div class="btn-main">
        <div class="btn">
            <button class="btn-hover color-9" onClick={this.startAsHost}>Create Game</button>
        </div>

        <div class="btn">
            <button class="btn-hover color-9" onClick={this.startAsPlayer}>Join Game </button>
        </div>
    </div>
</div>
        )
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        setHost: () => dispatch(setHost()),
        setPlayer: () => dispatch(setPlayer())
    }
}

export default connect(undefined, mapDispatchToProps)(DashboardPage);