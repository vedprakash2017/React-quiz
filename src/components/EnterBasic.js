import React from 'react';
import { connect } from 'react-redux';
import { socket } from '../app';
import { setRoom } from '../actions/game';
import { Redirect } from 'react-router-dom';
import Fade from 'react-reveal/Fade';
import AddQuestion from './addQuestion';
import axios from 'axios'

export class EnterBasic extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            room: "",
            questionCount: "5",
            error: "",
            background: "",
            changeNum: 1,
            changeNUM1:1
        }
    }
    sethim=()=>{
        console.log("hemlo");
    const multi = document.getElementsByClassName('multi')[0];
        console.log(multi);
    }

    onRoomChange = (e) => {
        const room = e.target.value;
        this.setState({ room });
    };

    onCountChange = (e) => {
        const questionCount = e.target.value;
        this.setState({ questionCount });
    }
    handleReset = (e) =>{

        if(this.state.room!='')
        {
        var mn = 0;
        axios.get('http://localhost:3005/api/v1/quiz/findRoom?room='+this.state.room)
        .then(res => {
            const num = res.data;
            mn = parseInt(num.persent);
            if(mn != 1)
        {
        const changeNum = 0;
        this.setState({ changeNum });
        }
        else
        {
        const changeNum1 = 0;
        this.setState({ changeNum1 });
            alert("Room Name already exist!")
        }
        }
        )
        
        }
        else
        {

        const changeNum1 = 0;

        this.setState({ changeNum1 });
            alert("Please Enter Room Name!")
        }
        // this.props.history.push("/addQuiz");
    }
    submitForm = (e) => {

        // this.props.history.push("/addQuiz");
        // e.preventDefault();
        // const config = {
        //     room: this.state.room,
        //     category: this.state.category,
        //     difficulty: this.state.difficulty,
        //     questionCount: this.state.questionCount
        // };
        // //console.log("submitting")
        // socket.emit("createRoom", config, (res) => {
        //     //console.log("res!", res);
        //     if (res.code === "success") {
        //         this.setState({ error: "" })
        //         this.props.setRoom(this.state.room);
        //         this.props.history.push("/lobby");
        //     } else {
        //         this.setState({ error: res.msg })
        //     }
        // });

    };

    render() {
        return (
            <div className="content-container" style={{height:"90vh"}}>
                {
                    this.props.type === "" && this.props.changeNUM1===1 && <Redirect to="/" />
                }
                
                
                {this.state.changeNum === 1 && <div className="box-layout__box" style={{ background: "rgba(0, 0, 0, 0.1)"}}> 
                    <Fade>
                        <form className="form" >
                        {/* onSubmit={this.submitForm} */}
                            <h1 className={"box-layout__title"}>Create New Game</h1>
                            {this.state.error && <p className="form__error">{this.state.error}</p>}
                            <input
                                type="text"
                                placeholder="Room Name"
                                autoFocus
                                value={this.state.room}
                                onChange={this.onRoomChange}
                                className="text-input"
                            />
                            <select className="select" value={this.state.questionCount} onChange={this.onCountChange}>
                                <option key="5" value="5">5 Questions</option>
                                <option key="10" value="10">10 Questions</option>
                                <option key="15" value="15">15 Questions</option>
                            </select>

                            
                            <div class="btn">
                                            <button class="btn-hover color-9" onClick={this.handleReset} type="button">Create</button>
                            </div>

                        </form>
                    </Fade>

                </div>
            }
            {this.state.changeNum === 0 && this.state.room != '' &&
            <div>
            
            <AddQuestion amount = {this.state.questionCount} roomno = {this.state.room}/>
            {/* <div onClick={this.sethim}>    
                <button> submit </button>
            </div> */}
           </div>
            }

            </div>
        )
    }
}


const mapStateToProps = (state) => ({
    categories: state.game.categories,
    type: state.type
});

const mapDispatchToProps = (dispatch) => ({
    setRoom: (room) => dispatch(setRoom(room))
});

export default connect(mapStateToProps, mapDispatchToProps)(EnterBasic);