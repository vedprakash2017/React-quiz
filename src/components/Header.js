import React from 'react';
import { socket } from '../app';
import { connect } from 'react-redux';
import { resetPlayers} from '../actions/players';
import { resetRoom, setQuestion, setMessage, resetGame } from '../actions/game';
import { resetType } from '../actions/clientType';


export class Header extends React.Component {
    
    handleClick = () => {
        socket.disconnect();
        socket.connect();
        this.props.resetPlayers();
        this.props.resetType();
        this.props.resetGame();
        this.props.history.push("/");
    }
    
    render() {
        return (
            <header className={"header"} style={{height:"8vh",background: "rgba(0, 0, 0, 0.05)"}}>
                <div className={"content-container"}>
                    <div className={"header__content"}>
                        <button className={"header__title button--link button"} onClick={this.handleClick}><h1>OpenTrivia</h1></button>
                        <button className="button--link button ">About</button>
                    </div>
                </div>
            </header>
        )
    }
};


const mapDispatchToProps = (dispatch) => ({
    resetPlayers: () => dispatch(resetPlayers()),
    resetType: () => dispatch(resetType()),
    resetGame: () => dispatch(resetGame())
});

export default connect(undefined, mapDispatchToProps)(Header);