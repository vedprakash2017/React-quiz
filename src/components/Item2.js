import React from 'react';
import { connect } from 'react-redux';
import RubberBand from 'react-reveal/RubberBand';

export const PlayerItem = (props) => (
<div>
    <div className="scoreboard">

    <div className="list-item">
        <h3>Player</h3>
        <h3>Score</h3>
    </div>

    {
                <div key={player.name} className="list-item">
                    <h3>{props.name}</h3>
                    <h3>{props.score}</h3>
                </div>
    }

    <div className="list-button">
        <button className="button" onClick={this.handleReset}>Start Again</button>
    </div>

    </div>
    <RubberBand>
        <div className="player__items">
            <h3 >{props.name} | {props.score}</h3>
        </div>
    </RubberBand>
    </div>
)

export default PlayerItem;

/*

{
type: 'ADD_PLAYER',
player: {name:"anushan", colour:"red"}
}

*/