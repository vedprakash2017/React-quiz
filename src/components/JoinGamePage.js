import React from 'react';
import { connect } from 'react-redux';
import { socket } from '../app';
import { Redirect } from 'react-router-dom';
import { setRoom } from '../actions/game';
import { HuePicker } from 'react-color';
import Fade from 'react-reveal/Fade';


export class JoinGamePage extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            room: "",
            name: "",
            colour: 1,
            error: ""
        }
    };

    onRoomChange = (e) => {
        const room = e.target.value;
        this.setState({ room });
    };

    onNameChange = (e) => {
        const name = e.target.value;
        this.setState({ name });
    };

    // handleChangeComplete = (color) => {
    //     this.setState({ colour: color.target.value });
    //     console.log(color.target.value);
    // };
    runParentHandleClick = (id) => {
        this.setState({ colour: id });
      }
    submitForm = (e) => {
        e.preventDefault();
        const config = {
           name: this.state.name,
           colour: this.state.colour,
           room: this.state.room 
        }

        socket.emit("joinRoom", config, (res) => {
            //console.log("res!", res);
            if (res.code === "success") {
                this.setState({ error: "" })
                this.props.setRoom(this.state.room);
                this.props.history.push("/lobby");
            } else {
                this.setState({ error: res.msg })
            }
        })
    }

    render() {
        return (
            <div className="content-container" style={{height:"90vh"}}>
                {
                    this.props.type === "" && <Redirect to="/" />
                }
                <div className="box-layout__box" style={{ background: "rgba(0, 0, 0, 0.1)"}}>
                    <Fade>
                        <form className="form" onSubmit={this.submitForm}>
                            <h1 className={"box-layout__title"}>Join Game</h1>
                            {this.state.error && <p className="form__error">{this.state.error}</p>}

                            <input
                                type="text"
                                placeholder="Room Name"
                                autoFocus
                                value={this.state.room}
                                onChange={this.onRoomChange}
                                className="text-input"
                            />

                            <input
                                type="text"
                                placeholder="User Name"
                                value={this.state.name}
                                onChange={this.onNameChange}
                                className="text-input"
                            />

                            <div className="form__picker">
                                    <div className="circular-portrait" onClick={() => this.runParentHandleClick('1')} value="1">
                                        <img src= {"https://edolve.s3.ap-south-1.amazonaws.com/" + 1+".png"} ></img>
                                    </div>
                                    <div className="circular-portrait" onClick={() => this.runParentHandleClick('2')} value="1">
                                        <img src={"https://edolve.s3.ap-south-1.amazonaws.com/" + 2+".png"} ></img>
                                    </div>
                                    <div className="circular-portrait" onClick={() => this.runParentHandleClick('3')} value="1">
                                        <img src={"https://edolve.s3.ap-south-1.amazonaws.com/" + 3+".png"} ></img>
                                    </div>
                                    <div className="circular-portrait" onClick={() => this.runParentHandleClick('4')} value="1">
                                        <img src={"https://edolve.s3.ap-south-1.amazonaws.com/" + 4+".png"} ></img>
                                    </div>
                                    <div className="circular-portrait" onClick={() => this.runParentHandleClick('5')} value="1">
                                        <img src={"https://edolve.s3.ap-south-1.amazonaws.com/" + 5+".png"} ></img>
                                    </div>
                                    <div className="circular-portrait" onClick={() => this.runParentHandleClick('6')} value="1">
                                        <img src={"https://edolve.s3.ap-south-1.amazonaws.com/" + 6+".png"} ></img>
                                    </div>
                                {/* <HuePicker
                                    color={this.state.colour}
                                    onChangeComplete={this.handleChangeComplete}
                                /> */}
                            </div>
                            <div class="btn">
                                            <button class="btn-hover color-9">Join</button>
                            </div>
                            {/* <button className="button">Join</button> */}



                        </form>

                    </Fade>
                </div>
            </div>
        );
    };

};

const mapStateToProps = (state) => ({
    type: state.type
});

const mapDispatchToProps = (dispatch) => ({
    setRoom: (room) => dispatch(setRoom(room))
});

export default connect(mapStateToProps, mapDispatchToProps)(JoinGamePage);