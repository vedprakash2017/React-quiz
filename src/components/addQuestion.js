import React from 'react';
import { connect } from 'react-redux';
import { socket } from '../app';
import { setRoom } from '../actions/game';
import { Redirect } from 'react-router-dom';
import Fade from 'react-reveal/Fade';
import QuestionNum from './QuestionNum';
import axios from 'axios';
export class addQuestion extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            room: this.props.roomno,
            questionCount: this.props.amount,
            error: "",
            background: "",
            amount:this.props.amount,
			changeNum1:1
        }
		this.handleSubmit = this.handleSubmit.bind(this);
    }

    componentDidMount(){
        console.log(this.props);
        //current position
var pos = 0;
//number of slides
var totalSlides = $('#slider-wrap ul li').length;
//get the slide width
var sliderWidth = $('#slider-wrap').width();


$(document).ready(function(){
	
	
	/*****************
	 BUILD THE SLIDER
	*****************/
	//set width to be 'x' times the number of slides
	$('#slider-wrap ul#slider').width(sliderWidth*totalSlides);
	
    //next slide 	
	$('#next').click(function(){
		slideRight();
	});
	
	//previous slide
	$('#previous').click(function(){
		slideLeft();
	});
	
	
	
	/*************************
	 //*> OPTIONAL SETTINGS
	************************/
	//automatic slider
	// var autoSlider = setInterval(slideRight, 3000);
	
	//for each slide 
	$.each($('#slider-wrap ul li'), function() { 
	   //set its color
	   var c = $(this).attr("data-color");
	   $(this).css("background",c);
	   
	   //create a pagination
	   var li = document.createElement('li');
	   $('#pagination-wrap ul').append(li);	   
	});
	
	//counter
	countSlides();
	
	//pagination
	pagination();
	
	//hide/show controls/btns when hover
	//pause automatic slide when hover
	$('#slider-wrap').hover(
	//   function(){ $(this).addClass('active'); clearInterval(autoSlider); }, 
	//   function(){ $(this).removeClass('active'); autoSlider = setInterval(slideRight, 3000); }
	  function(){ $(this).addClass('active');}, 
	  function(){ $(this).removeClass('active'); }
	);
	
	

});//DOCUMENT READY
	


/***********
 SLIDE LEFT
************/
function slideLeft(){
	pos--;
	if(pos==-1){ pos = totalSlides-1; }
	$('#slider-wrap ul#slider').css('left', -(sliderWidth*pos)); 	
	
	//*> optional
	countSlides();
	pagination();
}


/************
 SLIDE RIGHT
*************/
function slideRight(){
	pos++;
	if(pos==totalSlides){ pos = 0; }
	$('#slider-wrap ul#slider').css('left', -(sliderWidth*pos)); 
	
	//*> optional 
	countSlides();
	pagination();
}



	
/************************
 //*> OPTIONAL SETTINGS
************************/
function countSlides(){
	$('#counter').html(pos+1 + ' / ' + totalSlides);
}

function pagination(){
	$('#pagination-wrap ul li').removeClass('active');
	$('#pagination-wrap ul li:eq('+pos+')').addClass('active');
}
    }



	handleSubmit(event) {
		console.log("hmelo")
		// // Prevent default behavior
		event.preventDefault();
	
		const data = new FormData(event.target);
		for (let index = 1; index <= this.state.questionCount; index++) {


			const x = {};
			x.type = data.get('type-'+index);

			x.question = data.get('question-'+index);
					if(x.type == 'multiple' )
					{
						
						x.correct_answer = data.get('correct-'+index);
						const inc = [data.get('incorrect#1-'+index) , data.get('incorrect#2-'+index) , data.get('incorrect#3-'+index) ];
			
						if(x.question != "" && inc[0]!= "" && inc[1]!= "" && inc[2]!= "" && x.correct_answer!= "")
						{
						x.incorrect_answers = inc;
						// alert('Question added!');					
						}
						else
						{
							alert('Please fill all fields in question #'+index);
							return;
						}
					}
					else if(x.type == 'bool')
					{


						x.correct_answer =  data.get('bool-'+index);
						// console.log(x);
						if(x.question != undefined && x.correct_answer != undefined)
						{
							
						x.incorrect_answers = [!data.get('bool-'+index)];
							// alert('Question added!');	
						}
						else
						{
							alert('Please fill all fields in question #'+index);
							return;
						}
					}
			x.difficulty= 'easy';
			x.category = 'other';
			x.room = this.state.room;
			console.log(x);
			
			axios({
				method: "post",
				url: "http://localhost:3005/api/v1/quiz/add",
				data: JSON.stringify(x),
				headers: { "Content-Type": "application/json" },
			  })
				.then(function (response) {
				  //handle success
				//   console.log(response);
				})
				.catch(function (response) {
				  //handle error
				//   console.log(response);
				});
		}


        // const config = {
        //     room: this.state.room,
        //     category: 'other',
        //     difficulty: 'easy',
        //     questionCount: this.state.questionCount
        // };
        // //console.log("submitting")
        // socket.emit("createRoom", config, (res) => {
        //     //console.log("res!", res);
        //     if (res.code === "success") {
        //         this.setState({ error: "" })
        //         this.props.setRoom(this.state.room);
        //         this.props.history.push("/lobby");
        //     } else {
        //         this.setState({ error: res.msg })
        //     }
        // });




		// // Access FormData fields with `data.get(fieldName)`
		// // For example, converting to upper case
		// // data.set('username', data.get('username').toUpperCase());
		// console.log(data);
		// // Do your Axios stuff here
	  }



    render() {
        return (
            <div className="content-container" style={{height:"90vh"}}>



{
                    this.props.type === "" && this.state.changeNum1 == 1 && <Redirect to="/" />
                }
<div id="wrapper">
   
                
   	  <h1>Enter { this.props.amount } Questions </h1>
      <form onSubmit={this.handleSubmit}>  
   	  <div id="slider-wrap">
 
      	  <ul id="slider">

			<div style={{left:"0px",width:"100%"}}>    
            <QuestionNum props = {this.props.amount} />
      </div>
        
         </ul>
          <div class="btns" id="next"><i class="fa fa-arrow-right"></i></div>
          <div class="btns" id="previous"><i class="fa fa-arrow-left"></i></div>
          <div id="counter"></div>
          
          <div id="pagination-wrap">
            <ul>
            </ul>
          </div>
                 
      </div>

	  <div >
				<button type="submit" onclick="return false"> submit </button>
			</div>
	  </form>
   </div>


            </div>
        )
    }
}

export default addQuestion;

// export default connect(mapStateToProps, mapDispatchToProps)(CreateGamePage);