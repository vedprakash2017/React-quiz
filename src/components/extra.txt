style sheet
https://edolve.s3.ap-south-1.amazonaws.com/1620026386style.css



home audio
// https://edolve.s3.ap-south-1.amazonaws.com/1620131058home.wav
// https://edolve.s3.ap-south-1.amazonaws.com/1620131998home.ogg


{/* <ReactAudioPlayer src={"https://edolve.s3.ap-south-1.amazonaws.com/1620131998home.ogg"} controls ></ReactAudioPlayer> */}
 