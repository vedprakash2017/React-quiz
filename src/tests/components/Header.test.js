import React from 'react';
import {Header} from '../../components/Header';
import {shallow} from 'enzyme';


test("Should correctly render Header", () => {

    const wrapper = shallow(<Header startLogout={() => {}}/>);
    expect(wrapper).toMatchSnapshot();
});

test("Should call startLogout on button click", () => {
    const logout = jest.fn();
    const wrapper = shallow(<Header startLogout={logout}/>);
    wrapper.find("button").simulate("click");
    expect(logout).toHaveBeenCalled();
});
