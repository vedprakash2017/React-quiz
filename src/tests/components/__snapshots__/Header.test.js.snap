// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly render Header 1`] = `
<header
  className="header"
>
  <div
    className="content-container"
  >
    <div
      className="header__content"
    >
      <Link
        className="header__title"
        replace={false}
        to="/dashboard"
      >
        <h1>
          Boilerplate
        </h1>
      </Link>
      <button
        className="button--link button "
      >
        About
      </button>
    </div>
  </div>
</header>
`;
